db/migrate/20190708192323_fix_unserializable_notification_options.rb

```diff
@@ -0,0 +1,44 @@
+class FixUnserializableNotificationOptions < ActiveRecord::Migration[5.1]
+  class Notification < ActiveRecord::Base
+    belongs_to :subject, :polymorphic => true
+    belongs_to :notification_type, :class_name => "FixUnserializableNotificationOptions::NotificationType"
+  end
+
+  class NotificationType < ActiveRecord::Base
+  end
+
+  class CloudVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    Notification.where.not(:options => nil).includes(:subject).all.each do |n|
+      begin
+        # Options could have no longer available activerecord/activemodel classes from rails 5.0.0 -> 5.0.6, which will raise an ArgumentError on deserialization
+        # Fixing these to use 5.0.7+ smaller serialization format without all the private attributes is too hard, especailly since you can only read their YAML strings
+        # and make changes as a string, not as a Hash.
+        opts = YAML.load(n.options)
+      rescue ArgumentError
+        new_value = n.subject.try(:name) || n.subject.try(:description)
+
+        # We have no option but to blow away whatever was in options and set the subject information to what's in the associated subject object, even if it's been removed.
+        # puts "Notification: #{n.id} options are not loadable, setting options[:subject] from subject association: #{n.subject_type}: #{n.subject_id} with value: #{new_value}"
+
+        # Must use update_all to avoid callbacks and checks on this instance.
+        n.class.where(:id => n.id).update_all(:options => {:subject => new_value}.to_yaml)
```

Comment:
Would `n.update_column` work?  Does it make the database interaction better or worse?

Reply:
No, I tried many things on the instance but the dirty tracking blows up because it wants to evaluate if any attributes are changed.  Unfortunately, dirty tracking still occurs if you avoid callbacks.